src/ctx.rs

```diff
@@ -407,13 +400,14 @@ macro_rules! signed_to_unsigned {
 
 macro_rules! write_into {
     ($typ:ty, $size:expr, $n:expr, $dst:expr, $endian:expr) => {{
+        assert!($dst.len() >= $size);
```

Comment:
e.g., here too, this moving out of the unsafe block probably conflicts with other pr, but it's fine i think

Reply:
I moved it out of the unsafe block on purpose - to my knowledge, the unsafe should be as small as possible - limited to just the unsafe aspect of the code. I know it conflicts with the other PR, but I think we can easily fix those.  If ok, I would prefer to keep it, but up to you?

Reply:
sounds good